### DISCUSSION TITLE
[Custom Worker] Missing secret input text  & validateProperty

### DISCUSSION THREAD

**Discussion by rgdoms** [Keyfactor/ejbca-ce#683] [Q&A]
Hi again.

- The custom worker editing form lacks a 'text-secret' type field like the custom publisher has. I'm not sure why is that. I know that by using the CustomPublisherProperty.UI_TEXTINPUT_PASSWORD from Publisher, the field appears masked, but the content is still transmitted in plain text.  Am I missing something?

- It also seems that the custom worker fields do not have any validation methods unlike Custom Publishers have.

Tks

**Comment by primetomas** [comment]
Probably true. Workers is thought to operate on the local EJBCA system, using local api calls and such. There are no standard workers that calls remote services, so we didn't think about that a worker would need password storage.
In contrast to publishers that are designed to call to other systems, needing some sort of authentication.
